src/main/java/com/cloudbees/jenkins/support/util/CallAsyncWrapper.java

```diff
@@ -0,0 +1,30 @@
+package com.cloudbees.jenkins.support.util;
+
+import com.cloudbees.jenkins.support.SupportPlugin;
+import hudson.model.Computer;
+import hudson.remoting.Callable;
+import hudson.remoting.VirtualChannel;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public final class CallAsyncWrapper {
+
+    public static <V,T extends Throwable> hudson.remoting.Future<V> callAsync(final VirtualChannel channel, final Callable<V,T> callable) throws IOException {
+        Future<hudson.remoting.Future<V>> future = Computer.threadPoolForRemoting.submit(new java.util.concurrent.Callable<hudson.remoting.Future<V>>() {
+            @Override
+            public hudson.remoting.Future<V> call() throws IOException {
+                return channel.callAsync(callable);
+            }
+        });
+        try {
+            return future.get(SupportPlugin.REMOTE_OPERATION_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
```

Comment:
We may want to want to cancel the future on `TimeoutException` as nobody else is interested in the future after the exception is raised. Otherwise the task would still be executing:

```
        } catch (InterruptedException | ExecutionException e) {
            throw new IOException(e);
        }  catch (TimeoutException te) {
            future.cancel(true);
            throw new IOException(e);
        }
```

Would like maybe @MRamonLeon to confirm